src/main/java/com/brightpath/learnify/exception/mapper/ExceptionMapper.java

```diff
@@ -0,0 +1,57 @@
+package com.brightpath.learnify.exception.mapper;
+
+import com.brightpath.learnify.exception.authorization.UserNotAuthorizedToEditException;
+import com.brightpath.learnify.exception.authorization.UserNotAuthorizedToGetException;
+import com.brightpath.learnify.exception.badrequest.UpdatingQuizResultsFailedException;
+import com.brightpath.learnify.exception.notfound.ResourceNotFoundException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
+
+@ControllerAdvice
+public class ExceptionMapper {
+    private static final String BAD_REQUEST_DEFAULT_MESSAGE = "Bad request, reason: ";
+
+    //unauthorized
+    @ExceptionHandler
+    public ResponseEntity<?> onAuthenticationToGetFailedException(UserNotAuthorizedToGetException ex) {
+        return ResponseEntity
+                .status(UNAUTHORIZED)
```

Comment:
I know this is nitpicking but if user is not authorized to access resource but is authenticated then response code should be 403 FORBIDDEN instead of 401 UNAUTHORIZED

Reply:
right